### PR TITLE
[BugFix] Fix create agg meterialized view will make BE exit

### LINKED ISSUE #13182: Create materialized view make BE exit
### Steps to reproduce the behavior (Required)
1. CREATE TABLE
```
CREATE TABLE `test` (
  `dt` date NOT NULL COMMENT "时间分区",
  `a` bigint(20) NOT NULL COMMENT "",
  `b` bigint(20) NOT NULL COMMENT "",
  `c` bigint(20) NOT NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`dt`)
COMMENT "OLAP"
PARTITION BY RANGE(`dt`)
(PARTITION p20220501 VALUES [('2022-05-01'), ('2022-05-02')))
DISTRIBUTED BY HASH(`a`) BUCKETS 32
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false"
);
```
3. create materialized view 
```
CREATE MATERIALIZED VIEW test_view AS
select a,b from test group by b,a order by a;
```

### Expected behavior (Required)
report error or materialized view build

### Real behavior (Required)
BE exit with FATAL log

> F1108 20:21:33.801928 33001 column_aggregate_func.cpp:706] Check failed: false bad agg method NONE for column: b

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.3,2.2

### PR DESCRIPTION
**Pull request by mchades**
## What type of PR is this：
- [x] BugFix

## Which issues of this PR fixes ：

Fixes #13182

## Problem Summary(Required) ：

check the order-by columns must be identical to the group-by columns when create agg meterialized view

## Checklist:

- [x] I have added test cases for my bug fix or my new feature

### CODE REVIEW

**Comment by mchades** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStmt.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateMaterializedViewStmt.java [hunk 1]
```diff
@@ -517,7 +515,7 @@ private static void analyzeOrderByClause(CreateMaterializedViewStmt statement,
             throw new SemanticException("The number of columns in order clause must be less then " + "the number of "
                     + "columns in select clause");
         }
-        if (beginIndexOfAggregation != -1 && (orderByElements.size() != (beginIndexOfAggregation))) {
+        if (selectRelation.getGroupBy().size() != orderByElements.size()) {
```
Comment (luohaha) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/CreateMaterializedViewStmt.java:518]: Check the size of `group by` and `order by` can't fix all situation? We should check the column id instead.

**Comment by Astralidea** [comment]
Can you rebase this PR?

**Review by Astralidea — DISMISSED** [review]

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
You may need to look at how to make queryrewrite not report an error.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStmt.java|1|1|100.00%|[]|

**Comment by mchades** [comment]
> You may need to look at how to make queryrewrite not report an error.

ok, I have reused `supplyOrderColumn` to make up partial order columns. Pls take a review. @Astralidea

**Comment by Astralidea** [comment]
run pr-title-check

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Review by Astralidea — APPROVED** [review]

**Review by luohaha — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by Astralidea** [comment]
@Mergifyio backport branch-2.5

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-2.4

**Pull request merged**